src/plugins/dashboard/public/application/top_nav/show_share_modal.tsx

```diff
@@ -124,7 +122,9 @@ export function ShowShareModal({
     DashboardAppLocatorParams,
     'options' | 'query' | 'savedQuery' | 'filters' | 'panels' | 'controlGroupInput'
   > = {};
-  const unsavedDashboardState = dashboardSessionStorage.getState(savedDashboard.id);
+  const unsavedDashboardState = dashboardSessionStorage.getState(
+    currentDashboardState.savedObjectId
+  );
```

Comment:
Could we move line `144` up so that we can also use `savedObjectId` here instead?

```suggestion
  const unsavedDashboardState = dashboardSessionStorage.getState(savedObjectId);
```

Reply:
Fixed!